File: test/rs/case08_nested_call.rs
#[path = "../../src/rs_module/py_runtime.rs"]
mod py_runtime;
use py_runtime::{math_cos, math_exp, math_floor, math_sin, math_sqrt, perf_counter, py_bool, py_grayscale_palette, py_in, py_isalpha, py_isdigit, py_len, py_print, py_save_gif, py_slice, py_write_rgb_png};

// このファイルは自動生成です（native Rust mode）。

fn inc(mut x: i64) -> i64 {
    return ((x) + (1));
}

fn twice(mut x: i64) -> i64 {
    return inc(inc(x));
}

fn main() {
    py_print(twice(10));
}

File: test/cpp/case08_nested_call.cpp
#include "cpp_module/py_runtime.h"

// このファイルは `test/py/case08_nested_call.py` のテスト/実装コードです。
// 役割が分かりやすいように、読み手向けの説明コメントを付与しています。
// 変更時は、既存仕様との整合性とテスト結果を必ず確認してください。

int64 inc(int64 x) {
    return x + 1;
}

int64 twice(int64 x) {
    return inc(inc(x));
}

int main() {
    py_print(twice(10));
    return 0;
}
